src/bin/ion/commands/mod.rs

```diff
@@ -134,4 +152,96 @@ impl WithIonCliArgument for ClapCommand {
                 .help("Output format"),
         )
     }
+
+    fn with_compression_control(self) -> Self {
+        self.arg(
+            Arg::new("no-auto-decompress")
+                .long("no-auto-decompress")
+                .action(ArgAction::SetTrue)
+                .help("Turn off automatic decompression detection."),
+        )
+    }
```

Comment:
🗺️ By default, gzip and zstd compression are automatically handled in all commands. However, commands must opt-in to exposing any flag, including the one to disable decompression.